### ISSUE TITLE
crash at startup on current GIT on Debian 10

### ISSUE THREAD

**Issue by alaingdl** [gnudatalanguage/gdl#1305]
Compiled with the script `build_gdl.sh`

I got : 
```
(process:32802): GLib-GObject-WARNING **: 04:06:49.878: invalid (NULL) pointer instance

(process:32802): GLib-GObject-CRITICAL **: 04:06:49.878: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(process:32802): Gdk-CRITICAL **: 04:06:49.879: IA__gdk_screen_get_default_colormap: assertion 'GDK_IS_SCREEN (screen)' failed

(process:32802): Gdk-CRITICAL **: 04:06:49.879: IA__gdk_colormap_get_visual: assertion 'GDK_IS_COLORMAP (colormap)' failed

(process:32802): Gdk-CRITICAL **: 04:06:49.879: IA__gdk_screen_get_default_colormap: assertion 'GDK_IS_SCREEN (screen)' failed

(process:32802): Gdk-CRITICAL **: 04:06:49.879: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

(process:32802): Gdk-CRITICAL **: 04:06:49.879: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

(process:32802): Gdk-CRITICAL **: 04:06:49.879: IA__gdk_window_new: assertion 'GDK_IS_WINDOW (parent)' failed
./quick_start_GDL.sh : ligne 88 : 32802 Erreur de segmentation  /obs/alaingdl/GDL/gdl-1.0.1-git220409CMake/build/src/gdl

```

Yes, it is remotely, but `xclock` is OK

**Comment by alaingdl** [comment]
At the end, I can try on Ubuntu 20.04 and 18.04 with the script : 
@GillesDuvert @pjb7687 I hope you will find a fix !

* on 18.04, same kind on crash than in Debian 10

* on 20.04, I can start GDL but no display

```
*** PLPLOT ERROR, ABORTING OPERATION ***
plInitDispatchTable: Could not open drivers directory, aborting operation

(gdl:6212): Gtk-WARNING **: 22:37:45.950: Error loading theme icon 'text-x-generic' for stock: Icon 'text-x-generic' not present in theme elementary-xfce-darker
(gdl:6212): Gtk-WARNING **: 22:37:45.952: Error loading theme icon 'folder' for stock: Icon 'folder' not present in theme elementary-xfce-darker
(gdl:6212): Gtk-WARNING **: 22:37:45.952: Error loading theme icon 'folder' for stock: Icon 'folder' not present in theme elementary-xfce-darker
```

**Comment by Jeanne-jw** [comment]
For me, on Ubuntu 20.04.3 LTS, I can start and run GDL normally. 
But the same, on Debian GNU/Linux 10 in distance, I can't start GDL, with the same warnings : 
```
quick_start_GDL.sh: 11: [: !=: unexpected operator
Running current GDL with CMAKE_SOURCE_DIR/src/pro and CMAKE_SOURCE_DIR/testsuite in GDL_PATH
+/path/gdl/src/pro/:+/path/gdl/testsuite/

(process:32058): GLib-GObject-WARNING **: 10:27:11.558: invalid (NULL) pointer instance

(process:32058): GLib-GObject-CRITICAL **: 10:27:11.558: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(process:32058): Gdk-CRITICAL **: 10:27:11.558: IA__gdk_screen_get_default_colormap: assertion 'GDK_IS_SCREEN (screen)' failed

(process:32058): Gdk-CRITICAL **: 10:27:11.558: IA__gdk_colormap_get_visual: assertion 'GDK_IS_COLORMAP (colormap)' failed

(process:32058): Gdk-CRITICAL **: 10:27:11.558: IA__gdk_screen_get_default_colormap: assertion 'GDK_IS_SCREEN (screen)' failed

(process:32058): Gdk-CRITICAL **: 10:27:11.558: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

(process:32058): Gdk-CRITICAL **: 10:27:11.558: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

(process:32058): Gdk-CRITICAL **: 10:27:11.558: IA__gdk_window_new: assertion 'GDK_IS_WINDOW (parent)' failed
Segmentation fault
```

**Comment by GillesDuvert** [comment]
will hava a look asap

**Comment by GillesDuvert** [comment]
I can reproduce the related message is `gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed`
It is because DISPLAY is not set or not set as GTK likes it. Do not know what trick to use, as obviously wxWidgets does not handle it correctly.
GDL (development version) now install its own drivers (in share/gnudatalanguage/drivers)(mainly because of the recent plplot bugs), so I'm interested to know from @alaingdl what was the installation on 'on 20.04, I can start GDL but no display' above. Normally GDL should find the drivers **if GDL was installed** (not simply recompiled). But this may also come from the quick_start_GDL.sh procedure that is not maintained.

**Comment by jtappin** [comment]
Might be worth checking gdk-pixbuf versions. As there could be a need for a recent gdl and a recent gdk-pixbuf to trigger the problem.

**Comment by GillesDuvert** [comment]
Seems to stem from a conflict of libwx_baseu:
 **see #1312** 
when 2 libwx are present. (already seen this some time ago)
Probably related with build_gdl.sh that install libraries in a fixed order, suggesting it is not adequate with linux.
Suggestions welcome.

**Comment by GillesDuvert** [comment]
OK, solved in #1316 thanks to all
